pkg/splunk/enterprise/util.go

```diff
@@ -292,3 +342,53 @@ func DeleteOwnerReferencesForS3SecretObjects(client splcommon.ControllerClient,
 
 	return err
 }
+
+// GetAppListFromS3Bucket gets the list of apps from remote storage.
+func GetAppListFromS3Bucket(client splcommon.ControllerClient, cr splcommon.MetaObject, appFrameworkRef *enterprisev1.AppFrameworkSpec) (map[string]splclient.S3Response, error) {
+
+	scopedLog := log.WithName("GetAppListFromS3Bucket").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+
+	sourceToAppListMap := make(map[string]splclient.S3Response)
+
+	volMap := make(map[string]enterprisev1.VolumeSpec)
+
+	for _, vol := range appFrameworkRef.VolList {
+		volMap[vol.Name] = vol
+	}
```

Comment:
Nit: checkIfVolumeExists returns the index of the volume, if you want to use the slice directly, instead of going to a map.